app/src/main/java/com/infomaniak/mail/utils/Extensions.kt

```diff
@@ -73,6 +79,30 @@ fun <T> LiveData<T?>.observeNotNull(owner: LifecycleOwner, observer: (t: T) -> U
     observe(owner) { it?.let(observer) }
 }
 
+fun Recipient.displayedName(context: Context): String {
+    return if (AccountUtils.currentUser?.email == email) context.getString(R.string.contactMe) else getNameOrEmail()
+}
+
+fun Recipient.getNameOrEmail() = name?.ifBlank { email } ?: email
+
+fun <T> LiveData<T>.observeOnce(lifecycleOwner: LifecycleOwner, observer: Observer<T>) {
+    observe(lifecycleOwner, object : Observer<T> {
+        override fun onChanged(t: T?) {
+            observer.onChanged(t)
+            removeObserver(this)
+        }
+    })
+}
```

Comment:
Why ?

Reply:
It was previously used only by SwitchUserViewModel (and thus was private), but I needed it in another class, so I put it into Extension File as it is used in two different places

Reply:
done